src/modules/commander/Commander.cpp

```diff
@@ -1548,31 +1549,40 @@ Commander::run()
 		}
 
 
-		// Auto disarm when landed or kill switch engaged
+		// Auto disarm when landed or kill switch engaged and do not auto-disarm if launch detection is running
 		if (armed.armed) {
 
-			// Check for auto-disarm on landing or pre-flight
-			if (_param_com_disarm_land.get() > 0 || _param_com_disarm_preflight.get() > 0) {
+			if (_controller_status_sub.updated()) {
+				position_controller_status_s controller_status = {};
+				_controller_status_sub.copy(&controller_status);
+				launch_detection_running = controller_status.launch_detection_running;
+			}
```

Comment:
I wonder if we should also update this topic when not armed. That way we can reset the bool flag when disarmed, so it's not stale.

Reply:
@Kjkinney If you do it that way I think you could even write

`if (armed && ! launch_detection_running) {
...
} else {
...
}
`

Reply:
@Kjkinney Actually I noticed that @dagar suggested to do it this way.